usr.sbin/etcupdate/etcupdate.sh

```diff
@@ -219,13 +229,28 @@ build_tree()
 
 	# Purge auto-generated files.  Only the source files need to
 	# be updated after which these files are regenerated.
-	rm -f $1/etc/*.db $1/etc/passwd $1/var/db/services.db || return 1
+	autogenfiles="./etc/*.db ./etc/passwd ./var/db/services.db"
+	(cd $1 && printf '%s\n' $autogenfiles >> $metatmp && \
+	    rm -f $autogenfiles) || return 1
 
 	# Remove empty files.  These just clutter the output of 'diff'.
-	find $1 -type f -size 0 -delete || return 1
+	(cd $1 && find . -type f -size 0 -delete -print >> $metatmp) || \
+	    return 1
 
 	# Trim empty directories.
-	find $1 -depth -type d -empty -delete || return 1
+	(cd $1 && find . -depth -type d -empty -delete -print >> $metatmp) || \
+	    return 1
+
+	if [ -n "$noroot" ]; then
+		echo '#METALOG#' >> $metatmp || return 1
+		cat $1/METALOG >> $metatmp || return 1
+		awk '/^#METALOG#$/ { metalog = 1; next }
+		    { f=$1; gsub(/\/\/+/, "/", f) }
```

Comment:
This could do with a comment explaining what it's doing?

Reply:
Added a paragraph to the start of the if, unless you mean specifically this line that's just canonicalising // to / in the file names? (since the code to create the /usr/lib/debug/usr/tests tree uses /usr/lib/debug/${TESTSBASE}, and TESTSBASE includes a leading /, which should be fixed, but also want this to not be fragile when people invariably screw that up somewhere else in future)